llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -961,23 +961,27 @@ getSizeFromDwarfHeaderAndSkip(BinaryStreamReader &Reader) {
 
 /// Given a list of MCFragments, return a vector with the concatenation of their
 /// data contents.
-/// If any fragment is not an MCDataFragment, an error is returned.
+/// If any fragment is not an MCDataFragment, or the fragment is an
+/// MCDwarfLineAddrFragment and the section containing that fragment is not a
+/// debug_line section, an error is returned.
 static Expected<SmallVector<char, 0>>
-mergeDataFragmentContents(const MCSection::FragmentListType &FragmentList) {
+mergeMCFragmentContents(const MCSection::FragmentListType &FragmentList,
+                        bool IsDebugLineSection = false) {
   SmallVector<char, 0> mergedData;
-  for (const MCFragment &Fragment : FragmentList) {
-    if (const auto *DataFragment = dyn_cast<MCDataFragment>(&Fragment))
-      llvm::append_range(mergedData, DataFragment->getContents());
+  for (const MCFragment &F : FragmentList) {
```

Comment:
In general, it is better to reduce the diff to present reviewers with a minimal patch, I'm a bit confused as to what caused the renaming

Reply:
This is my bad, I had a similar function and when resolving the merge conflict I just went with F instead of Fragment